### PR TITLE
Simplify before testing convergence

### LINKED ISSUE #16699: Wrong result in the convergence of infinite sums with factorials
```
>>> n = Symbol('n', integer = True)
>>> expr = factorial(n) / factorial(n+2)
>>> s = Sum(expr, (n, 1, oo))
>>> print(s.is_convergent())
False
```

The correct answer is `True`

### PR DESCRIPTION
**Pull request by AnimeshSinha1309**
Added simplification of function, before testing for convergence and added tests.

#### References to other Issues or PRs
Fixes #16699

#### Brief description of what is fixed or changed
Fixing issues with Sum of the factorial series, just added simplification of the expression before the tests for convergence start.

#### Release Notes

* concrete
  * simplify sums before testing for convergence

### CODE REVIEW

**Comment by smichr** [comment]
Is there a way to just return None if you know that simplification? Or if this is a high-level routine then maybe using simplify makes sense.

**Comment by maurogaravello** [comment]
I do not think this resolves the issue. This patch works in the specific example and probably is better than nothing.
If you consider 
```
expr = 1 - cos(1/n)
s = Sum(expr, (n, 1, oo))
```
the result given by SymPy is `False`, even with the proposed patch, but the correct result is `True`.
This is due, as in the original example, to a non sharp result of order.

A possible alternative is to evaluate `limit(expr * n, n, oo)`. In the case the result is a positive number (not 0), then the convergence is `False`. This is in substitution of the lines (445-446)
```
if p1_series_test[p] >= -1:
                return S.false 
```

**Comment by AnimeshSinha1309** [comment]
@smichr I did not fully understand what you said. Do you mean that if the call to simplify fails? When exactly should I return None.

My concern with calling simplify here was that the code might get slowed down, I think that this is becoming one of the slowest tests in the test suite. I am will try a smarter workaround if that is a problem, possibly changing the `__new__()` function in the Order class.

**Comment by AnimeshSinha1309** [comment]
Also the following calls to the order function still don't give answer that make sense:
```python
>>> print(O(factorial(n), (n, oo)))
O(factorial(n), (n, oo))
>>> print(O(factorial(n)/factorial(n-2), (n, oo)))
O(1, (n, oo))
>>> print(O(factorial(n)/factorial(n+2), (n, oo)))
O(1, (n, oo))
```
I would want that the `factorial()` calls all mapped to a result `O(exp(n*log(n)), (n, oo))`, because the factorial is bounded by `(n/2)**n` and `(n/3)**n`.

**Comment by AnimeshSinha1309** [comment]
So again, this function call is the problem here, right?
```python
>>> n = Symbol('n')
>>> O(1-cos(n), (n, 0))
O(1)
```
The result should have been `O(n^2)`, but that would require the Order function to have some understanding of the Taylor series expansions, won't it?

Also calling `limit(expr * n)` might just work, but even then all the calls to order don't really make much sense. :cry:

**Comment by maurogaravello** [comment]
> So again, this function call is the problem here, right?
> 
> ```python
> >>> n = Symbol('n')
> >>> O(1-cos(n), (n, 0))
> O(1)
> ```

Exactly. This is the main problem. 
`O(1)` here is correct, but not sharp!
`O(n**-2)` is also correct. This would produce the correct result.

**Comment by smichr** [comment]
I see that I had an incom plete thought in my suggestion. What it was intended to suggest is that maybe obtaining a meaningless O expression of something like that would allow you to know that you should return none of else use a limit or simplification.

**Comment by oscarbenjamin** [comment]
What's the status of this? @AnimeshSinha1309 @smichr

**Comment by AnimeshSinha1309** [comment]
Sorry, I did not follow up on this. My Attempts did not work out, I would appriciate any help I can get. I am available to work again, trying now.

**Comment by AnimeshSinha1309** [comment]
Honestly I am out of ideas, simple fixes to the `is_convergent()` method, or calling Simplify, fail on examples like `1-cos(1/n)`, and fixing the Order constructor is hard and I always end up breaking a bunch of tests. I am wondering if it would be okay to just fix a few cases, and leave fixing the Order function for another issue.

**Comment by AnimeshSinha1309** [comment]
https://github.com/sympy/sympy/blob/12d8f5354db591ba6600c50f4188b7b973a859fc/sympy/concrete/products.py#L359-L360
Is this example not incorrect, since `Product(n/(n + 1), (n, 1, oo)) = (1/2) * (2/3) * ... * (n-1/n) = 1/n = 1/oo = 0`. So the answer should be True, it's convergent, should it not?

**Comment by jksuom** [comment]
An infinite product with limit 0 is considered divergent.

**Comment by AnimeshSinha1309** [comment]
Thanks for that, I fixed my mistake, undid the example that I had changed. I am just confused about one more thing, I changed a few lines in Summation, and added 3 tests for it, why is my Codecov score still falling, should I just keep adding mode examples?

File: sympy/concrete/summations.py [hunk 1]
```diff
@@ -496,12 +513,6 @@ def is_convergent(self):
         except NotImplementedError:
             pass
 
-        ### ------------- alternating series test ----------- ###
-        dict_val = sequence_term.match((-1)**(sym + p)*q)
-        if not dict_val[p].has(sym) and is_decreasing(dict_val[q], interval):
-            return S.true
```
Comment (oscarbenjamin) [inline sympy/concrete/summations.py]: It looks like this and everything below is untested. Are these tests (alternating series etc) not already implemented somewhere else?

Comment (AnimeshSinha1309) [reply sympy/concrete/summations.py]: I don't fully understand, there are already tests for Alternating Series and for everything else that follows. I believe that the issue might be that my p-series limit comparison is just returning a solution, blocking access to some of the functions later in the code. But now the codecov score looks pretty decent, right? And if not, what is it that I fix, I have no clue about codecov?

**Comment by czgdp1807** [comment]
Closing as the associated issue is outdated.

**Pull request closed without merging**